tests/functional/Glpi/Asset/Capacity/HasSoftwaresCapacity.php

```diff
@@ -0,0 +1,310 @@
+ */
+
+namespace tests\units\Glpi\Asset\Capacity;
+
+use DisplayPreference;
+use Entity;
+use Glpi\Asset\Asset;
+use Glpi\Tests\CapacityTestCase;
+use Item_SoftwareVersion;
+use Log;
+use Software;
+use SoftwareVersion;
+
+class HasSoftwaresCapacity extends CapacityTestCase
+{
+    /**
+     * Get the tested capacity class.
+     *
+     * @return string
+     */
+    protected function getTargetCapacity(): string
+    {
+        return \Glpi\Asset\Capacity\HasSoftwaresCapacity::class;
+    }
+
+    public function testCapacityActivation(): void
+    {
+        global $CFG_GLPI;
+
+        $root_entity_id = getItemByTypeName(Entity::class, '_test_root_entity', true);
+
+        $superadmin_p_id = getItemByTypeName(\Profile::class, 'Super-Admin', true);
+        $profiles_matrix = [
+            $superadmin_p_id => [
+                READ   => 1,
+                CREATE => 1,
+            ],
+        ];
+
+        $definition_1 = $this->initAssetDefinition(
+            capacities: [
+                \Glpi\Asset\Capacity\HasSoftwaresCapacity::class,
+                \Glpi\Asset\Capacity\HasSoftwaresCapacity::class,
+            ],
+            profiles: $profiles_matrix
+        );
+        $classname_1  = $definition_1->getAssetClassName();
+        $definition_2 = $this->initAssetDefinition(
+            capacities: [
+                \Glpi\Asset\Capacity\HasHistoryCapacity::class,
+            ]
+        );
+        $classname_2  = $definition_2->getAssetClassName();
+        $definition_3 = $this->initAssetDefinition(
+            capacities: [
+                \Glpi\Asset\Capacity\HasSoftwaresCapacity::class,
+                \Glpi\Asset\Capacity\HasHistoryCapacity::class,
+            ],
+            profiles: $profiles_matrix
+        );
+        $classname_3  = $definition_3->getAssetClassName();
+
+        $has_capacity_mapping = [
+            $classname_1 => true,
+            $classname_2 => false,
+            $classname_3 => true,
+        ];
+
+        foreach ($has_capacity_mapping as $classname => $has_capacity) {
+            // Check that the class is globally registered
+            if ($has_capacity) {
+                $this->array($CFG_GLPI['software_types'])->contains($classname);
+            } else {
+                $this->array($CFG_GLPI['software_types'])->notContains($classname);
+            }
+
+            // Check that the corresponding tab is present on items
+            $item = $this->createItem($classname, ['name' => __FUNCTION__, 'entities_id' => $root_entity_id]);
+            $this->login(); // must be logged in to get tabs list
+            if ($has_capacity) {
+                $this->array($item->defineAllTabs())->hasKey('Item_SoftwareVersion$1');
+            } else {
+                $this->array($item->defineAllTabs())->notHasKey('Item_SoftwareVersion$1');
+            }
+        }
+    }
+
+    public function testCapacityDeactivation(): void
```

Comment:
You should create `Item_SoftwareLicense` relations and related display preferences and test their presence/deletion as for the `Item_SoftwareVersion` relations and related display preferences.